### PR TITLE
Redux form

### PR DESCRIPTION
**Pull request by Hellsingi**
Add layout with redux-form

### CODE REVIEW

File: my-project/src/app/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class App extends Component {
+    submit = values => {
+        alert(JSON.stringify(values));
+    };
+    render() {
+        return (
+            <>
+                <h1 className="title">Sing in</h1>
+                <LoginForm onSubmit={this.submit}></LoginForm>
```
Comment (JuliaAlaeva) [inline my-project/src/app/index.js]: It isn't necessary closing tag here:
`<LoginForm onSubmit={this.submit} />`

File: my-project/src/components/Field/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import './style.scss';
+
+const myInput = (props) => {
+    const { input, type, placeholder, meta } = props;
+    return (
+        <>
+            <input {...props.input} type={props.type} placeholder={props.placeholder} />
+            {meta.error &&
```
Comment (JuliaAlaeva) [inline my-project/src/components/Field/index.js]: It's better to use descructurization for avoid duplicate `meta.`:
`const myInput = (input, type, placeholder, meta: { error, touched })`

File: my-project/src/components/Field/style.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+input {
```
Comment (JuliaAlaeva) [inline my-project/src/components/Field/style.scss]: Better called style file `index.scss`/`field.scss`.

File: my-project/src/components/LoginForm/style.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
+.image{
```
Comment (JuliaAlaeva) [inline my-project/src/components/LoginForm/style.scss]: Use gap after selector during all file.
`.image {`

File: my-project/src/app/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+import LoginForm from '../components/LoginForm/index'
+import './style.scss';
+
+class App extends Component {
+    submit = values => {
```
Comment (JuliaAlaeva) [inline my-project/src/app/index.js]: It's better to create this method in `<LoginForm />`.

File: my-project/src/components/Field/index.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import './style.scss';
+
+const myInput = (props) => {
+    const { input, type, placeholder, meta } = props;
+    return (
+        <>
+            <input {...props.input} type={props.type} placeholder={props.placeholder} />
```
Comment (JuliaAlaeva) [inline my-project/src/components/Field/index.js]: There these props in const input/type/placeholder, use them here.

File: my-project/src/components/LoginForm/index.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                        <Field
+                            name="password"
+                            component={myInput}
+                            type="password"
+                            placeholder="Password"
+                            validate={requiredInput}
+                        />
+                        <div>
+                            <button type="button" onClick={reset}>Clean form</button>
+                            <button type="submit" label="submit">Submit</button>
```
Comment (JuliaAlaeva) [inline my-project/src/components/LoginForm/index.js]: What is an attribute `label`?

**Review by JuliaAlaeva — COMMENTED** [review]
1. Name your file/folder in the same way (lowercase/uppercase);
2. Move common components to 'common' folder in 'components' folder;

File: my-project/src/components/Field/index.js (hunk 2 above)
Comment (Hellsingi) [reply my-project/src/components/Field/index.js]: done

File: my-project/src/app/index.js (hunk 1 above)
Comment (Hellsingi) [reply my-project/src/app/index.js]: done

File: my-project/src/components/LoginForm/style.scss (hunk 4 above)
Comment (Hellsingi) [reply my-project/src/components/LoginForm/style.scss]: done

File: my-project/src/app/index.js (hunk 5 above)
Comment (Hellsingi) [reply my-project/src/app/index.js]: so / so

File: my-project/src/components/Field/index.js (hunk 6 above)
Comment (Hellsingi) [reply my-project/src/components/Field/index.js]: done

File: my-project/src/components/LoginForm/index.js (hunk 7 above)
Comment (Hellsingi) [reply my-project/src/components/LoginForm/index.js]: done

File: my-project/src/components/Field/style.scss (hunk 3 above)
Comment (Hellsingi) [reply my-project/src/components/Field/style.scss]: so / so

**Pull request merged**